crates/toml_edit/src/de/value.rs

```diff
@@ -21,12 +25,20 @@ use crate::de::Error;
 ///     name: String,
 /// }
 ///
-/// let value = r#"{ title = 'TOML Example', owner = { name = 'Lisa' } }"#;
-/// let deserializer = value.parse::<toml_edit::de::ValueDeserializer>().unwrap();
-/// let config = Config::deserialize(deserializer).unwrap();
+/// let toml_str = r#"{ title = 'TOML Example', owner = { name = 'Lisa' } }"#;
 ///
+/// let deserializer = toml_str
+///     .parse::<toml_edit::de::ValueDeserializer>()
+///     .unwrap();
+/// let config = Config::deserialize(deserializer).unwrap();
 /// assert_eq!(config.title, "TOML Example");
 /// assert_eq!(config.owner.name, "Lisa");
+///
+/// let toml_item: toml_edit::Item = toml_str.parse().unwrap();
+/// let another_deserializer = toml_item.into_value().unwrap().into_deserializer();
+/// let another_config = Config::deserialize(another_deserializer).unwrap();
+/// assert_eq!(another_config.title, "TOML Example");
+/// assert_eq!(another_config.owner.name, "Lisa");
```

Comment:
This should have a bit more narrative explaining whats going on, whether thats making them two separate examples with a bridge statement or a comment in them.

Or we can drop the second example